packages/roosterjs-content-model-plugins/lib/autoFormat/link/createLinkAfterSpace.ts

```diff
@@ -1,37 +1,53 @@
+import { splitTextSegment } from '../../pluginUtils/splitTextSegment';
 import type { IEditor } from 'roosterjs-content-model-types';
 
 /**
  * @internal
  */
 export function createLinkAfterSpace(editor: IEditor) {
-    editor.formatContentModel(model => {
+    editor.formatContentModel((model, context) => {
         const selectedSegmentsAndParagraphs = getSelectedSegmentsAndParagraphs(
             model,
             false /* includingFormatHolder */
         );
-        if (selectedSegmentsAndParagraphs[0] && selectedSegmentsAndParagraphs[0][1]) {
-            const length = selectedSegmentsAndParagraphs[0][1].segments.length;
-            const marker = selectedSegmentsAndParagraphs[0][1].segments[length - 1];
-            const textSegment = selectedSegmentsAndParagraphs[0][1].segments[length - 2];
-            if (
-                marker.segmentType == 'SelectionMarker' &&
-                textSegment.segmentType == 'Text' &&
-                !textSegment.link
-            ) {
-                const link = textSegment.text.split(' ').pop();
-                if (link && matchLink(link)) {
-                    textSegment.text = textSegment.text.replace(link, '');
-                    const linkSegment = createText(link, marker.format, {
-                        format: {
-                            href: link,
-                            underline: true,
-                        },
-                        dataset: {},
-                    });
-                    selectedSegmentsAndParagraphs[0][1].segments.splice(length - 1, 0, linkSegment);
-                    return true;
+        if (selectedSegmentsAndParagraphs.length > 0 && selectedSegmentsAndParagraphs[0][1]) {
+            const markerIndex = selectedSegmentsAndParagraphs[0][1].segments.findIndex(
+                segment => segment.segmentType == 'SelectionMarker'
+            );
+            const paragraph = selectedSegmentsAndParagraphs[0][1];
+            if (markerIndex > 0) {
+                const textSegment = paragraph.segments[markerIndex - 1];
+                const marker = paragraph.segments[markerIndex];
+                if (
+                    marker.segmentType == 'SelectionMarker' &&
+                    textSegment &&
+                    textSegment.segmentType == 'Text' &&
+                    !textSegment.link
+                ) {
+                    const link = textSegment.text.split(' ').pop();
+                    const url = link?.trim();
+
+                    if (url && link && matchLink(url)) {
+                        const linkSegment = splitTextSegment(
+                            textSegment,
+                            paragraph,
+                            textSegment.text.length - link.trimLeft().length,
+                            textSegment.text.trimRight().length
+                        );
+                        linkSegment.link = {
+                            format: {
+                                href: url,
```

Comment:
Actually, this may generate a link that is not workable. Because the protocol part is necessary in the link. You need to retrieve url string from the result of matchLink().

To repro, type something "www.bing.com", space, then export the content and try to click the link, it won't open.